runtime/compiler/control/JITServerHelpers.cpp

```diff
@@ -883,17 +883,25 @@ JITServerHelpers::romMethodOfRamMethod(J9Method* method)
    }
 
 void
-JITServerHelpers::postStreamFailure(OMRPortLibrary *portLibrary, TR::CompilationInfo *compInfo)
+JITServerHelpers::postStreamFailure(OMRPortLibrary *portLibrary, TR::CompilationInfo *compInfo, bool retryConnectionImmediately)
    {
    OMR::CriticalSection postStreamFailure(getClientStreamMonitor());
 
    OMRPORT_ACCESS_FROM_OMRPORT(portLibrary);
    uint64_t current_time = omrtime_current_time_millis();
-   if (!_waitTimeMs)
-      _waitTimeMs = TR::Options::_reconnectWaitTimeMs;
-   if (current_time >= _nextConnectionRetryTime)
-      _waitTimeMs *= 2; // Exponential backoff
-   _nextConnectionRetryTime = current_time + _waitTimeMs;
+   if (retryConnectionImmediately)
+      {
+      _nextConnectionRetryTime = current_time;
+      }
+   else
+      {
+      if (!_waitTimeMs)
+         _waitTimeMs = TR::Options::_reconnectWaitTimeMs;
+      if (current_time >= _nextConnectionRetryTime)
+         _waitTimeMs *= 2; // Exponential backoff
+      _nextConnectionRetryTime = current_time + _waitTimeMs;
+      }
+
```

Comment:
Suggestion: print the value of "retryConnectionImmediately" in the verbose message below.